src/mobile/datetime-picker/index.ts

```diff
@@ -0,0 +1,412 @@
+    this._dateConverted = dateValueConverter(this._dateAndTime.date);
+    const isValidValue =
+      validateDateTimeValue(this._dateAndTime.date, this._dateAndTime.time) &&
+      isValidDate(this._dateConverted);
+    if (!isValidValue) {
+      throwErrorAfterUpdateComplete(this, FORMAT_IS_NOT_VALID);
+      return false;
+    }
+
+    return true;
+  }
+
+  willUpdate(_changedProperties: PropertyValues): void {
+    const changeByUI = this._changeDateByUI || this._changeTimeByUI;
+    if (changeByUI) {
+      this._updateValueAndErrorWhenUIChange();
+      return;
+    }
+    this._updateValueWhenSetter();
+  }
+
+  update(changedProperties: PropertyValues) {
+    if (changedProperties.has("value")) {
+      if (this.value === undefined) {
+        this._setUndefinedValue();
+      }
+      if (this.value === "") {
+        this._setEmptyValue();
+      }
+    }
+    super.update(changedProperties);
+  }
+
+  private _updateValueWhenSetter() {
+    this._errorText = this.error;
+    if (this.value === "" || this.value === undefined) {
+      this._previousTimeValue = "";
+      this._errorFormat = "";
+      return;
+    }
+    this._setDateTimeValueSeparate(this._dateAndTime, this._dateConverted);
+    this.value = this._getDateTimeString();
+  }
+
+  private _setDateTimeValueSeparate(dateTime: DateAndTime, dateValue: string) {
+    this._dateValue = dateValue;
+    this._timeValue =
+      this._dateValue && isValidDate(dateValue)
+        ? timeValueConverter(dateTime.time.slice(0, 5))
+        : this._previousTimeValue;
+  }
+
+  private _updateValueAndErrorWhenUIChange() {
+    const validFormat = this._checkDateTimeFormat();
+    this.value = validFormat ? this.value : undefined;
+    this._errorText = validFormat ? this.error : this._errorFormat;
+  }
+
+  private _checkDateTimeFormat() {
+    const isMissingDatePart = Boolean(this._timeValue) && !this._dateValue;
+    const isMissingTimePart = Boolean(this._dateValue) && !this._timeValue;
+    const validFormat =
+      !this._errorFormat && !isMissingDatePart && !isMissingTimePart;
+    return validFormat;
+  }
+
+  private _setUndefinedValue() {
+    if (this._changeTimeByUI) return;
+
+    if (this._errorFormat) {
+      if (this._changeDateByUI) return;
+
+      this._dateValue = "";
+      this._timeValue = "";
+      return;
+    }
+    this._dateValue = this._previousDateValue;
+    this._timeValue = this._previousTimeValue;
+  }
+
+  private _setEmptyValue() {
+    this._dateValue = "";
+    this._timeValue = "";
+    this._previousTimeValue = "";
+    this._previousDateValue = "";
+    this._errorFormat = "";
+  }
+
+  private _getDateTimeValue(value: string | undefined) {
+    if (value === "" || value === undefined) return { date: "", time: "" };
+
+    const dateTime = value.split("T");
+    const date = dateTime[0];
+    const time = dateTime[1];
+    if (value.indexOf("T") === value.length - 1 || dateTime.length > 2)
```

Comment:
what happen when value.indexOf("T") === -1 ?

Reply:
If the value don't have "T" the time part will be "00:00"